cvat/apps/dataset_manager/tests/test_rest_api_formats.py

```diff
@@ -1517,16 +1516,16 @@ def _clear(*_, file_path: str, file_ctime: str):
             ):
                 mock_rq_get_current_job.return_value = MagicMock(timeout=5)
 
-                set_condition(clear_process_has_run)
-                file_is_been_used_error_raised = False
+                file_is_being_used_error_raised = False
                 try:
+                    set_condition(clear_process_has_run)
```

Comment:
At this point you still can't say the function has been called. It would be correct in the except block below or after it, or in a side effect after the lock acquisition attempt (need to catch the `FileIsBeingUsedError` exception there, actually, so it should be a wrapper).

Reply:
>  It would be correct in the except block below or after it

Then there will be deadlock because the export process waits with an acquired lock when `clear_process_has_run` is set. At the same time, the clear process raises `FileIsBeingUsedError` only after it acquires a lock and checks that a file should not be deleted.

> or in a side effect after the lock acquisition attempt

I've tried to wrap the lock.acquire method to add a side effect and set `clear_process_has_run`, but this approach has 2 problems:
1. it's a bad idea to patch a shared lock that is used by another process from a  _clear process
2. in such a case, the second process will not be able to acquire a lock after releasing the first process

Reply:
> Then there will be deadlock because the export process waits with an acquired lock when clear_process_has_run is set. At the same time, the clear process raises FileIsBeingUsedError only after it acquires a lock and checks that a file should not be deleted.

I mean, yes, the cleanup function is expected to wait for the export, if both export and cleanup functions actually use locks. The deadlock is broken by the lock acquisition timeout exception. The `FileIsBeingUsedError` should not be thrown in this scenario. This is how the test worked before these changes. Now this exception can correctly occur after the export process checked the file and before the new export is finished. But the point of the test was to check that the exported file exists after the export, even if cleanup runs immediately after the export.

I think, maybe it should not even be changed, or it can be changed to check 2 cases: if the export exited after file check and the cleanup function runs exactly after this (now it's possible) and the previous case (as nothing changed there).